third_party/terraform/tests/resource_compute_instance_test.go.erb

```diff
@@ -4750,3 +4797,148 @@ resource "google_compute_resource_policy" "foo" {
 
 `, instance, instance, suffix)
 }
+
+func testAccComputeInstance_subnetworkChange(suffix, instance string) string {
+	return fmt.Sprintf(`
+	data "google_compute_image" "my_image" {
+		family  = "debian-9"
+		project = "debian-cloud"
+	}
+
+	resource "google_compute_network" "inst-test-network" {
+		name = "tf-test-network-%s"
+	}
+
+	resource "google_compute_network" "inst-test-network2" {
+		name = "tf-test-network2-%s"
+	}
+
+	resource "google_compute_subnetwork" "inst-test-subnetwork" {
+		name          = "tf-test-compute-subnet-%s"
+		ip_cidr_range = "10.0.0.0/16"
+		region        = "us-east1"
+		network       = google_compute_network.inst-test-network.self_link
```

Comment:
Does .id work here and for some of the other self_link references? We're trying to move towards always referring to other resources by their id when possible (and filing issues when it turns out it isn't), since most newer APIs don't use self links and we don't want people to have to remember what field to reference.

Reply:
does work